compiler-core/src/error.rs

```diff
@@ -406,54 +412,98 @@ impl Error {
             }
         }
 
-        Self::DependencyResolutionFailed(match error {
+        match error {
             ResolutionError::NoSolution(mut derivation_tree) => {
                 derivation_tree.collapse_no_versions();
 
                 let mut conflicting_packages = HashSet::new();
                 collect_conflicting_packages(&derivation_tree, &mut conflicting_packages);
 
-                wrap_format!("Unable to find compatible versions for \
-the version constraints in your gleam.toml. \
-The conflicting packages are:
+                let conflict_names: Vec<EcoString> = conflicting_packages
+                    .iter()
+                    .map(|pkg| (*pkg).to_string().into())
+                    .collect();
+
+                let locked_conflicts: Vec<EcoString> = conflict_names
+                    .iter()
+                    .filter(|name| locked.contains_key(*name))
+                    .cloned()
+                    .collect();
+
+                if locked_conflicts.is_empty() {
+                    Error::DependencyResolutionFailed {
+                            error: format!(
+                                "Unable to find compatible versions for the version constraints in your gleam.toml.\n\
+                                 The conflicting packages are:\n{}",
+                                conflicting_packages.into_iter().map(|s| format!("- {s}")).join("\n")
+                            ),
+                            locked_conflicts,
+                    }
+                } else {
+                    Error::DependencyResolutionFailed {
+                            error: format!(
+                                "Unable to find compatible versions for the version constraints in your gleam.toml.\n\
+                                 The conflicting packages are:\n{}",
+                                locked_conflicts.iter().map(|s| format!("- {s}")).join("\n")
```

Comment:
Why are only some of the conflicting packages are listed in this case? Wouldn't we want to show them all? If not we need to update the message to say this is only a subset of them.